### PR TITLE
Compile with -Werror -Wall -Wextra and fix associated warnings

### PR DESCRIPTION
**Pull request by tklengyel**
The changes are mechanic fixing of warnings issued mostly by -Wunused-parameter and -Wunused-variable

### CODE REVIEW

File: libvmi/driver/kvm/kvm.c [hunk 1]
```diff
@@ -1115,11 +1115,12 @@ kvm_get_memory_patch(
     }
     else {
         // get the data from kvm
-        nbytes =
-            read(kvm_get_instance(vmi)->socket_fd, buf, length + 1);
-        if (nbytes != (length + 1)) {
+        nbytes = read(kvm_get_instance(vmi)->socket_fd, buf, length + 1);
+        if ( nbytes <= 0 )
```
Comment (smaresca) [inline libvmi/driver/kvm/kvm.c]: What if nbytes = length/2 ? Was the original condition intended to check that data of proper size was returned, or just for an invalid rc from read()?

Comment (smaresca) [inline libvmi/driver/kvm/kvm.c]: nevermind, I see the condition below now

File: libvmi/performance.c [hunk 2]
```diff
@@ -53,52 +53,6 @@ print_measurement(
            (long int) ktv_end.tv_usec, *diff);
 }
 
-static double
```
Comment (smaresca) [inline libvmi/performance.c]: No harm in keeping these around, no?

File: libvmi/driver/kvm/kvm.c (hunk 1 above)
Comment (tklengyel) [inline libvmi/driver/kvm/kvm.c]: Yea, the problem was comparing signed and unsigned variables but just casting it to unsigned is not safe without first checking if it's less then 0.

**Comment by smaresca** [comment]
I like tagging deliberately unused parameters in the function prototype. 

GCC/Clang 
http://stackoverflow.com/a/12891181 

MSCV
#define UNUSED(x) __pragma(warning(suppress: 4100 4101)) x

File: libvmi/performance.c (hunk 2 above)
Comment (tklengyel) [inline libvmi/performance.c]: They break compiling with -Werror so if we want to keep them we can just comment them out.

**Comment by tklengyel** [comment]
Retest this please.

**Comment by tklengyel** [comment]
@smaresca: sure, looks fine to me that way too.

**Comment by tklengyel** [comment]
Retest this please.

**Pull request merged**